transport.go

```diff
@@ -98,9 +113,47 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 
 		wv.setWebhookDefaults(&webhook, r.RemoteAddr)
 
+		auth, present := bascule.FromContext(c)
+		var partners []string
+		if !present {
+			return nil, errAuthNotPresent
+		}
+		if auth.Token == nil {
+			return nil, errAuthTokenIsNil
+		}
+
+		switch auth.Token.Type() {
+		case "basic":
+			authHeader := r.Header[config.basicPartnerIDsHeader]
+
+			for _, value := range authHeader {
+				fields := strings.Split(value, ",")
+				for i := 0; i < len(fields); i++ {
+					fields[i] = strings.TrimSpace(fields[i])
+				}
+				partners = append(partners, fields...)
+			}
+		case "jwt":
+			authToken := auth.Token
+			partnersInterface, attrExist := bascule.GetNestedAttribute(authToken.Attributes(), basculechecks.PartnerKeys()...)
+			if !attrExist {
+				return nil, errPartnerIDsDoNotExist
+			}
+			vals, err := cast.ToStringSliceE(partnersInterface)
+			if err != nil {
+				return nil, fmt.Errorf("%w: %v", errGettingPartnerIDs, err)
+			}
+			partners = vals
+		default:
+			return nil, errAuthIsNotOfTypeBasicOrJWT
+		}
```

Comment:
we could make this its own helper function. Something like `extractPartnerIDs` which returns both a slice and an error type?